src/Pos/Wallet/Web/Server/Methods.hs

```diff
@@ -461,22 +478,38 @@ sendExtended sendActions cpassphrase srcCAddr dstCAddr c curr title desc = do
                 -- TODO: this should be removed in production
                 let txHash = hash tx
                 whenJust mRems $ \(_, remCAddr) -> do
-                    () <$ addHistoryTx remCAddr curr title desc (THEntry txHash tx True Nothing)
+                    () <$ addHistoryTx (caaAddress remCAddr) curr title desc (THEntry txHash tx True Nothing)
                     removeAccount srcCAddr
                 () <$ addHistoryTx dstCAddr curr title desc (THEntry txHash tx False Nothing)
-                addHistoryTx srcCAddr curr title desc (THEntry txHash tx True Nothing)
+                addHistoryTx (caaAddress srcCAddr) curr title desc (THEntry txHash tx True Nothing)
+  where
+    -- for [CSM-185]
+    selectSrcAccounts :: WalletWebMode ssc m => Coin -> [CAccountAddress] -> m (Coin, [CAccountAddress])
+    selectSrcAccounts reqCoins accounts = case accounts of
+        _ | reqCoins <= mkCoin 0 ->
+            throwM $ Internal "Spending non-positive amount of money!"
+        [] ->
```

Comment:
I think this can be written more short, clear and elegant using guards and PatternGuards